src/components/Fetch/index.js

```diff
@@ -523,6 +551,10 @@ componentWillReceiveProps(nextProps){
       <Col lg={1}><ToggleButton value={this.state.wrap || false} onToggle={(value) => {this.setState({wrap: !value})}} /></Col>
       <Col componentClass={ControlLabel} lg={2}>Case Sensitive</Col>
       <Col lg={1}><ToggleButton value={this.state.caseSensitive || false} onToggle={this.toggleCaseSensitive.bind(this)} /></Col>
+      <Col componentClass={ControlLabel} lg={2}>Filter Intersection</Col>
+      <Col lg={1}><ToggleButton value={this.state.filterIntersection || false} onToggle={this.toggleFilterIntersection.bind(this)} /></Col>
```

Comment:
I think this control would be clearer with the states labeled as OR/AND (and the logic remaining the same as what you have). Something like:
![image](https://user-images.githubusercontent.com/29384396/38153373-64cd1330-343a-11e8-98bd-8a3398c388cf.png)